### PR TITLE
Feature/deferred task correlation

### PR DESCRIPTION
**Pull request by dshafer**
- [x] Update CHANGELOG.md
- [ ] Update README.md (as needed)
- [ ] New dependencies were added to `pyproject.toml`
- [x] `pip install` succeeds with a clean virtualenv
- [x] There are new or modified tests that cover changes
- [x] Test coverage is maintained or expanded

### CODE REVIEW

File: tests/machinery/test_base.py [hunk 1]
```diff
@@ -820,6 +820,39 @@ def setUp(self):
...
+        conf = tasks.future_task_runner.app.conf
+        old_val = conf["task_always_eager"]
+        conf["task_always_eager"] = True
+
+        old_set_cid = tasks._set_correlation_id
+        old_get_cid = tasks._get_correlation_id
+        tasks._set_correlation_id = set_correlation_id
+        tasks._get_correlation_id = lambda: "cid-sentinel"
+        try:
+
```
Comment (smholloway) [inline tests/machinery/test_base.py:838]: 💄 extra blank line

File: tests/machinery/test_base.py [hunk 2]
```diff
@@ -820,6 +820,39 @@ def setUp(self):
             "filtered_retry_count_2": 0,
         }
 
+    def test_future_task_runner_sets_cid(self):
```
Comment (smholloway) [inline tests/machinery/test_base.py:823]: Do we have a test covering when `cid` is not present?

Comment (dshafer) [reply tests/machinery/test_base.py:823]: basically every existing test is checking that condition

Comment (smholloway) [reply tests/machinery/test_base.py:823]: Sounds good. Thanks!

**Review by smholloway — APPROVED** [review]

**Comment by bgrant** [comment]
Can you add a CHANGELOG entry @dshafer ?

File: django_declarative_apis/machinery/tasks.py [hunk 3]
```diff
@@ -26,8 +26,10 @@
     import cid.locals
```
Comment (bgrant) [inline django_declarative_apis/machinery/tasks.py:26]: Since we use it in testing, shoudl we add `django-cid` to our `optional-dependencies`?

Comment (dshafer) [reply django_declarative_apis/machinery/tasks.py:26]: we're actually mocking it in testing, so it's not necessary.  What's the rule for putting things in `optional-dependencies`?  It doesn't seem to simplify installing things with pip

Comment (bgrant) [reply django_declarative_apis/machinery/tasks.py:26]: Oh- maybe you overload the necessary method in testing.  I'm trying to think about where, if anywhere, `django-cid` should go in our listed dependencies in `pyproject.toml`.  Maybe nowhere, and (ideally) we would just document that DDA supports it somewhere (README or docs).  That doesn't need to be in this PR.

**Review by bgrant — APPROVED** [review]
LGTM!

**Pull request merged**